airflow/providers/amazon/aws/hooks/glue_crawler.py

```diff
@@ -0,0 +1,216 @@
+        super().__init__(*args, **kwargs)
+
+    @cached_property
+    def glue_client(self):
+        """:return: AWS Glue client"""
+        return self.get_conn()
+
+    def check_iam_role(self, role_name: str) -> str:
+        """
+        Checks if the input IAM role name is a
+        valid pre-existing role within the caller's AWS account.
+        Is needed because the current Boto3 (<=1.16.46)
+        glue client create_crawler() method misleadingly catches
+        a non-existing role as a role trust policy error.
+
+        :param role_name: IAM role name
+        :type role_name: str
+        :return: IAM role name
+        """
+        iam_client = self.get_client_type('iam', self.region_name)
+
+        iam_client.get_role(RoleName=role_name)
+
+    def has_crawler(self, crawler_name) -> bool:
+        """
+        Checks if the crawler already exists
+
+        :param crawler_name: unique crawler name per AWS account
+        :type crawler_name: str
+        :return: Returns True if the crawler already exists and False if not.
+        """
+        self.log.info("Checking if AWS Glue crawler already exists: %s", crawler_name)
+
+        try:
+            self.glue_client.get_crawler(Name=crawler_name)
+            return True
+        except self.glue_client.exceptions.EntityNotFoundException:
+            return False
+
+    def update_crawler(self, **crawler_kwargs) -> str:
+        """
+        Updates crawler configurations
+
+        :param crawler_kwargs: Keyword args that define the configurations used for the crawler
+        :type crawler_kwargs: any
+        :return: True if crawler was updated and false otherwise
+        """
+        crawler_name = crawler_kwargs['Name']
+        current_crawler = self.glue_client.get_crawler(Name=crawler_name)['Crawler']
+
+        update_config = {
+            key: value for key, value in crawler_kwargs.items() if current_crawler[key] != crawler_kwargs[key]
+        }
+        if update_config != {}:
+            self.log.info("Updating crawler: %s", crawler_name)
+            self.glue_client.update_crawler(**crawler_kwargs)
+            self.log.info("Updated configurations: %s", update_config)
+            return True
+        else:
+            return False
+
+    def create_crawler(self, **crawler_kwargs) -> str:
+        """
+        Creates an AWS Glue Crawler
+
+        :param crawler_kwargs = Keyword args that define the configurations used to create the crawler
+        :type crawler_kwargs = any
+        :return: Name of the crawler
+        """
+        crawler_name = crawler_kwargs['Name']
+        self.log.info("Creating AWS Glue crawler: %s", crawler_name)
+
+        try:
+            glue_response = self.glue_client.create_crawler(**crawler_kwargs)
+            return glue_response['Crawler']['Name']
+        except self.glue_client.exceptions.InvalidInputException as general_error:
+            self.check_iam_role(crawler_kwargs['Role'])
```

Comment:
i am not sure why you would want to check iam role if you're gonna raise anyway?

Reply:
As mentioned in the `check_iam_role()` docs, the Glue API's `create_crawler()` function doesn't catch non-existing roles properly and throws an inaccurate trust policy error. So the purpose of the`check_iam_role()` is to throw the IAM client's non-existing role error instead of the inaccurate trust policy error. 

But your comment does remind me that I should change the `check_iam_role()` to throw an AirflowException with the non-existing role error so omits the `AirflowException(general_error)`

Reply:
i remember this discussion

i would leave it with the original error.

i think adding this only adds confusion.

my 2 cents 🤦‍♂️

Reply:
Any thoughts on this? @mik-laj @mschmo @feluelle

Reply:
my intention is not to block this pr if no one else cares about this but just want to highlight the issue in case anyone does

Reply:
I would suggest to add @marshall7m comment there, sth. like:

```python
try:
    ...
except self.glue_client.exceptions.InvalidInputException:
    # <add-explanation-here>
    self.check_iam_role(crawler_kwargs['Role'])
    raise
```
I would not raise `AirflowException` tbh. Personally I don't like using this type of exception if it actually is unrelated to airflow. Just re-raise the error in case it has nothing to do with the iam role. If it's because of the iam role it `check_iam_role` will raise a more accurate error.

WDYT?

Reply:
my thought is, why create a method `check_iam_role` that serves no purpose other than throwing an error, when the only reason you're calling it is because you _already know_ the role is bad.

in testing we saw that when InvalidInputException is raised due to role, it mentions the role in the error.  so it seems pointless to me.

but, it's also not really going to kill me if this goes in so i'll quit yapping about it after this :)

Reply:
> in testing we saw that when InvalidInputException is raised due to role, it mentions the role in the error. so it seems pointless to me.

Got it. If that's really the case then we do not need and probably should not check the iam role again. We save a request :) Only if it does not mention the role it makes sense.

Reply:
After thinking about it, removing the `check_iam_role` method is the right move. As you said @dstandish, the InvalidInputException points the error to the role so that exception should be enough. If the user tries to debug why they got the trust policy error regarding the role, they would probably check if the role they're checking is the right role :).  I'm glad you brought it up, it was a good learning point for me 👍 .

Reply:
@feluelle Thanks for the input, it's always nice to minimize the number of requests :) 

I'll push the changes sometime today